association.go

```diff
@@ -277,45 +302,56 @@ func (a *Association) sendCookieEcho() error {
 
 // Close ends the SCTP Association and cleans up any state
 func (a *Association) Close() error {
+	a.log.Debugf("[%s] closing association..", a.name())
+
+	// TODO: shutdown sequence
+	a.lock.Lock()
```

Comment:
Is it possible/worth making `setState` an atomic, the less locking the better (IMO)

Reply:
```
func (a *Association) setState(state associationState) {
	if a.state != state {
		a.log.Debugf("[%s] state change: '%s' => '%s'", a.name(), a.state.String(), state.String())
		a.state = state
	}
}
```
I found logging before and after state is useful.  Caller is assume to hold the mutext in the current code, so not much gain.  I will add a comment to say, caller must hold the lock instead!

Reply:
Fixed at ef7368e